tests/test_dataclasses.py

```diff
@@ -1622,6 +1622,30 @@ class D2:
     ]
 
 
+def test_base_dataclasses_annotations_resolving(create_module):
+    # Have to use a string due to the __future__ import
```

Comment:
I think it'd be better to use `d1: 'OddInt'` so that the future import isn't needed.